src/driver/xen.rs

```diff
@@ -29,6 +33,24 @@ pub struct Xen {
     back_ring: vm_event_back_ring,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum XenDriverError {
+    #[error("event version mismatch: {0} <-> {1}")]
+    EventVersionMismatch(u32, u32),
+    #[error(transparent)]
```

Comment:
IIRC this duplicates the error message from the source error and therefore makes walking error source chains awkward.

Reply:
So we should replace `#[error(transparent)]` by `#[error("defined error message")]` ?

Reply:
okay, done